### ISSUE TITLE
Thank you

### ISSUE THREAD

**Issue by ramnes** [realh/roxterm#1]
Hey @realh,

I'm too lazy to make an account on SourceForge but after reading that you want to drop ROXTerm support, I had to write a small message.

Thank you a billion time for all your work on ROXTerm. I've never used ROX but this has been my favorite terminal emulator for years now, and I'm very sad to imagine it disappear.

I've used daily xterm, gnome-terminal, terminator, urxvt, and probably others I can't remember. And then ROXTerm. Its the only one that has at the same time few dependencies, a good overall speed, and that's  easily configurable.

On a side note, if you're hoping for someone here on GitHub to take the lead, I'd recommend you to at least push the tags, and migrate the issues that are still open on SourceForge to GitHub (https://github.com/cmungall/gosf2github might help). I'd be happy to take a look at it and see if I can do anything then.

Thanks again.

**Comment by realh** [comment]
OK, I've started importing all the old issues and sent you a collaborator invite. Thanks for the offer of help.
There may be some extra info in the SF discussions from people who have found partial fixes. I also wrote an experimental neovim gui last year, which solves a similar problem (text grids of a defined size in a GtkNotebook), so I could offer a summary of the way that works, but I didn't get around to adding the full functionality that roxterm has, and it doesn't use geometry hints, so I don't know whether roxterm can be fixed without removing that feature.

**Comment by ramnes** [comment]
Wow, that's a bunch of issues. Maybe we could start by removing all the outdated and/or non-reproducible ones.

What are the issues that you feel are the most critical?

**Comment by ramnes** [comment]
Also, are you on IRC or something?

**Comment by realh** [comment]
Yes, it looks like a lot, but you're right, most of them are obsolete. Not that there aren't a lot of genuine problems, and I'm pretty sure geometry problems go quite a bit deeper than just that shrinking window bug. I feel that tackling that from the other end - ie basically starting with a new application and adding in roxterm's features using old code where possible - is more appealing.

But for starting a new GUI application now, C doesn't cut it any more. C++ makes it more bearable, but not great. Vala would be great, except it's clearly heading into oblivion. Python is the only other realistic choice at the moment, but a lot of people seem to think that roxterm is lean and "fast", and python would put them off even if this performance is largely imaginary or irrelevant. And even with type annotations I'd prefer a compiled language with stricter typing. If the G* community can't keep Vala afloat they need to adopt an externally backed language like C# or Kotlin and put a big effort into gir bindings, but they don't seem interested. :-(

Oh yeah, another thing that should probably be done is to replace the build system. Meson would be best I think.

I'm not really "on" IRC, but I can fire up a session, it would be good to have a chat about this stuff. Do you regularly use a particular channel or anything?

**Comment by realh** [comment]
In fact, it looks like the migration tool copied **all** the old issues, not just the open ones, and didn't mark them as closed here :-(. That's a nuisance.

**Comment by ramnes** [comment]
I've just joined #roxterm on irc.freenode.net, let's talk there.

**Comment by ramnes** [comment]
As discussed on IRC, the main issue to solve is #126.

**Comment by realh** [comment]
I looked at David's patch. It seems to be based on a considerably older version of roxterm (before I got rid of a lot of #ifdefs supporting legacy GTK 2.x), but I made some changes based on it and it seems to work. There's still a problem, because if you look at the debug output there are some negative values in the geometry hints, which leads to them just being ignored AFAICT, so it might be best just to stop using set_geometry_hints altogether if we can't fix that.

I spent most of the evening dealing with terminal size preservation ie resizing the window to fit the terminal when adding/removing the menu bar and tab bar etc, when GTK's default behaviour is the other way round. That was a pain.

Meanwhile, the fix for #126 has inexplicably caused a new bug: when there's only one tab its label is stupidly small.
